Dockerfile

```diff
@@ -15,29 +12,56 @@ RUN cd /root/bcc && \
     /usr/lib/pbuilder/pbuilder-satisfydepends && \
     PARALLEL=$(nproc) ./scripts/build-deb.sh release
 
-FROM ubuntu:bionic
+RUN cd /root && git clone https://github.com/libbpf/libbpf.git && \
+    cd /root/libbpf && git checkout 030ff87857090ae5c9d74859042d05bfb3b613a2  && cd src && mkdir build root && BUILD_STATIC_ONLY=y OBJDIR=build DESTDIR=root make install
+
+FROM ubuntu:20.04 as compiler
+ENV DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get update && \
-    apt-get install -y --no-install-recommends git build-essential libelf1 software-properties-common
+    apt-get install -y --no-install-recommends git build-essential pkg-config libelf-dev libz-dev software-properties-common clang
 
 RUN add-apt-repository ppa:longsleep/golang-backports && \
     apt-get install -y --no-install-recommends golang-1.17-go
 
 ENV PATH="/usr/lib/go-1.17/bin:$PATH"
 
 COPY --from=builder /root/bcc/libbcc_*.deb /tmp/libbcc.deb
+COPY --from=builder /root/bcc/libbpf-tools  /tmp/libbpf-tools
+COPY --from=builder /root/libbpf/ /tmp/libbpf
 
+RUN cd /tmp/libbpf/src && make install
 RUN dpkg -i /tmp/libbcc.deb
 
 COPY ./ /go/ebpf_exporter
 
-RUN cd /go/ebpf_exporter && \
-    GOPROXY="off" GOFLAGS="-mod=vendor" go install -v -ldflags=" \
+RUN cd /go/ebpf_exporter && CGO_CFLAGS="-I /usr/include/bpf" CGO_LDFLAGS="-lbpf" GOPATH="" GOPROXY="off" GOFLAGS="-mod=vendor" go install -v -ldflags=" \
     -X github.com/prometheus/common/version.Version=$(git describe) \
     -X github.com/prometheus/common/version.Branch=$(git rev-parse --abbrev-ref HEAD) \
     -X github.com/prometheus/common/version.Revision=$(git rev-parse --short HEAD) \
     -X github.com/prometheus/common/version.BuildUser=docker@$(hostname) \
     -X github.com/prometheus/common/version.BuildDate=$(date --iso-8601=seconds) \
     " ./cmd/ebpf_exporter
+RUN cd /go/ebpf_exporter/examples/CORE && make all
+
+
+FROM ubuntu:20.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends build-essential pkg-config libelf-dev libz-dev software-properties-common
+
+COPY --from=builder /root/bcc/libbcc_*.deb /tmp/libbcc.deb
+COPY --from=builder /root/bcc/libbpf-tools  /tmp/libbpf-tools
+COPY --from=builder /root/libbpf/ /tmp/libbpf
+COPY --from=compiler /root/go/bin/ebpf_exporter /ebpf_exporter
+COPY --from=compiler /go/ebpf_exporter/examples /examples
+
+RUN cd /tmp/libbpf/src && make install
+RUN dpkg -i /tmp/libbcc.deb
+RUN find /tmp -name "*.git*"|xargs rm -Rf
+RUN find /examples -name "*.git*"|xargs rm -Rf
 
-RUN /root/go/bin/ebpf_exporter --version
+ENV LD_LIBRARY_PATH=/lib64
+CMD /ebpf_exporter  --web.listen-address=:10678  --config.file=/examples/CORE/config.yaml
```

Comment:
As mentioned above, this image is not intended for direct consumption, so we don't need this line.

There's also a standard port allocated for ebpf_exporter:

* https://github.com/prometheus/prometheus/wiki/Default-port-allocations